addons/pos_pinelabs/static/src/app/payment_pinelabs.js

```diff
@@ -0,0 +1,245 @@
+        return this._callPinelabs(data, "pinelabs_cancel_payment_request").then((response) => {
+            if (response.errorMessage === "CANNOT CANCEL AS TRANSACTION IS IN PROGRESS") {
+                this.pos.notification.add(
+                    _t("Your cancellation request is in the queue and will be processed shortly.")
+                );
+                clearTimeout(this.pollingTimeout);
+                setTimeout(async () => {
+                    const response = await this._callPinelabs(
+                        data,
+                        "pinelabs_cancel_payment_request"
+                    );
+                    if (response.responseCode) {
+                        this.pos.notification.add(_t("Error: %s", response.errorMessage), {
+                            title: _t("Warning"),
+                            type: "danger",
+                        });
+                    }
+                }, this.paymentInitialTime + CANCEL_REQUEST_TIME_LIMIT - Date.now());
+                return Promise.resolve(true);
+            }
+            if (response.errorMessage) {
+                this._showError(response.errorMessage);
+                return Promise.resolve(false);
+            }
+            this._pinelabsHandleResponse(response);
+            return Promise.resolve(true);
+        });
+    }
+
+    _processPinelabs() {
+        const order = this.pos.get_order();
+        const line = order.get_selected_paymentline();
+        const sequenceNumber = order.payment_ids.filter(
+            (pi) => pi.payment_method_id.use_payment_terminal === "pinelabs"
+        ).length;
+        if (line.amount < 0) {
+            this._showError(_t("Cannot process transactions with negative amount."));
+            return Promise.resolve();
+        }
+
+        const orderId = order?.pos_reference?.replace(" ", "").replaceAll("-", "").toUpperCase();
+        const referencePrefix = this.pos.config.name.replace(/\s/g, "").slice(0, 4);
+        localStorage.setItem(
+            "transactionNumber",
+            referencePrefix + "/" + orderId + "/" + crypto.randomUUID().replaceAll("-", "")
+        );
+        const data = {
+            amount: line.amount * 100, // We need to provide the amount in paisa since Pine Labs processes amounts in paisa.
+            transactionNumber: localStorage.getItem("transactionNumber"),
+            sequenceNumber: sequenceNumber, // In the case of multiple transactions for the same order, it is important to follow the correct sequence of transactions.
+        };
+        this.paymentInitialTime = Date.now();
+        return this._callPinelabs(data, "pinelabs_make_payment_request").then((data) => {
+            return this._pinelabsHandleResponse(data);
+        });
+    }
+
+    /**
+     * Polling
+     * This method uses the pinelabsFetchPaymentStatus function to check the payment status
+     * every 10 seconds until it is resolved.
+     */
+    async _waitForPaymentToConfirm() {
```

Comment:
why don't you use a webhook,  like we do for adyen ? 
https://developer.pinelabs.com/in/instore/cloud-integration#:~:text=APPROVED%22%2C%0A%7D-,8.%20Post%20Back%20URL%3A,-In%20this%20approach
![image](https://github.com/user-attachments/assets/f7b840c5-137c-4c7a-92ba-cf4518f5dab2)

Reply:
With Adyen, we support both online and terminal payments. In contrast, with Pine Labs, we are integrating only terminal payments. Using a webhook configuration similar to Adyen for terminal payments in Pine Labs makes the process more complicated. Pinelabs provides simple status requests to get payment status.

Why do we complicate the use of payment terminals with webhook configuration?

Reply:
why is a webhook more complicated ?

Reply:
we have this pattern in a lot of places and it proved to be very convenient. Adyen online payments are not part of pos. The webhook logic from `pos_adyen` is only related to terminal payments. It's not like for adyen "it was worth doing it" because we have more usecases. We have it because it's more straight forward than polling

Reply:
We have a Razorpay terminal where we’ve implemented a similar setup. There are two scenarios for using the terminal payment:

1. We simply need to add the credentials, and the terminal is ready for use.
2. We need to set up the entire webhook configuration.

From the customer's perspective, which option is easier to use?

Reply:
isn't there a way to let pinelabs know the url of the webhook programatically? Like we have in `pos_urban_piper` with the `_register_webhook` method

Reply:
The documentation does not provide an API for registering webhooks.

Reply:
> The documentation does not provide an API for registering webhooks.

Can you please explain in the commit message the steps that are needed to test this feature? This way we will have a reference in case anyone has to test this. 

( you can link to resources ( pinelabs website, odoo docs, etc ), but please provide extracts from there ( links might get stale) )